commit-status-publisher-server/src/main/java/jetbrains/buildServer/commitPublisher/stash/StashPublisher.java

```diff
@@ -214,4 +229,58 @@ private String getUsername() {
   private String getPassword() {
     return myParams.get(Constants.STASH_PASSWORD);
   }
+
+  private StashCommit findPullRequestCommit(@NotNull SBuild build) throws PublisherException {
+    String apiUrl = myParams.get(Constants.STASH_BASE_URL);
+    String projectKey = myParams.get(Constants.STASH_PROJECT_KEY);
```

Comment:
For other publishers, as well as for StashSettings::testConnection method we use GitRepositoryParser.parseRepository method to extract the repo details necessary to build the URL. It may be beneficial to introduce an explicit Commit Status Publisher settings for the same, but they have to be optional in my opinion.